packages/itwin/ec3-widget/src/components/LabelAction.tsx

```diff
@@ -0,0 +1,274 @@
+
+    if (availableStringColumns.indexOf(itemName) === -1 && options.length !== 0)
+      setItemName("");
+
+    return options;
+  }, [availableStringColumns, itemName]);
+
+  const getStringColumnOptions = ((material: string | undefined) => {
+    const options = stringColumnOptions
+      .filter((x) => !materials.some((m) => m.nameColumn === x.label))
+      .filter((x) => x.label !== itemName);
+
+    if (material)
+      options.push({ label: material, value: material });
+    return options;
+  });
+
+  const numericalColumnOptions = useMemo(() => {
+    const options = availableNumericalColumns.map((col) => ({
+      label: col,
+      value: col,
+    }));
+
+    if (availableNumericalColumns.indexOf(itemQuantity) === -1 && options.length !== 0)
+      setItemQuantity("");
+
+    return options;
+  }, [availableNumericalColumns, itemQuantity]);
+
+  const addPair = (() => {
+    const pair: Material = {
+      nameColumn: undefined,
+    };
+    const newMaterials = materials.map((x) => { return { nameColumn: x.nameColumn }; });
+    newMaterials.push(pair);
+    setMaterials(newMaterials);
+  });
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (label) {
+      setName(label.name);
+      setItemName(label.elementNameColumn);
+      setItemQuantity(label.elementQuantityColumn);
+      setMaterials(label.materials.map((x) => { return { nameColumn: x.nameColumn }; })); // creating a copy of array, so original (in the parent) isn't modified
+    } else {
+      setItemName("UserLabel");
+    }
+  }, [label]);
+
+  const onChangeCallback = useCallback(async (table: string, numCols: string[], strCols: string[]) => {
+    if (table !== reportTable) {
+      setMaterials([{ nameColumn: undefined }]);
+      setName(table);
+    }
+    setReportTable(table);
+    setNumericalColumns(numCols);
+    setStringColumns(strCols);
+  }, [reportTable]);
+
+  return (
+    <>
+      <WidgetHeader
+        title={label?.name ?? "Label"}
+        returnFn={goBack}
+        disabled={isLoading}
+      />
+      <div className='ec3w-label-details-container'>
+        <Fieldset legend='Label' className='ec3w-label-details'>
+          <Small className='ec3w-label-field-legend'>
+            Asterisk * indicates mandatory fields.
+          </Small>
+
+          <ReportTableSelector
+            selectedReportTable={reportTable}
+            template={template}
+            placeHolder={isLoading ? "Loading report tables" : "Select report table"}
+            onChange={onChangeCallback}
+            setLoading={setIsLoading}
+          />
+
+          <LabeledInput
+            id='name'
+            name='name'
+            label='Name'
+            value={name}
+            onChange={(event) => {
+              setName(event.target.value);
+            }}
+          />
+
+          <div className="ec3w-dropdown-select-container">
+            <div className="ec3w-dropdown-select-combo-box">
+              <Label htmlFor="combo-input" required>
```

Comment:
Looks like a `LabeledSelect`.

Reply:
Changed to LabeledSelect

Reply:
Are the divs surounding the `LabeledSelect` still required? How does the component look with and without the divs?

Reply:
Looks exactly the same. Removed the divs.